packages/react-dom/src/__tests__/ReactErrorBoundaries-test.internal.js

```diff
@@ -576,103 +578,148 @@ describe('ReactErrorBoundaries', () => {
     };
   });
 
-  it('does not swallow exceptions on mounting without boundaries', () => {
+  it('does not swallow exceptions on mounting without boundaries', async () => {
     let container = document.createElement('div');
-    expect(() => {
-      ReactDOM.render(<BrokenRender />, container);
-    }).toThrow('Hello');
+    let root = ReactDOMClient.createRoot(container);
+    await expect(async () => {
+      await act(async () => {
+        root.render(<BrokenRender />);
+      });
+    }).rejects.toThrow('Hello');
```

Comment:
I've added `.rejects` before `.toThrow` because this test is async now and this tells jest to handle the rejected promise correctly